### DISCUSSION TITLE
The problems of the saveAll of input

### DISCUSSION THREAD

**Discussion by babyfish-ct** [babyfish-ct/jimmer#410] [Ideas]
In Jimmer, saving entity and saving input are different.

```
<E> SimpleSaveResult<E> save(E Entity);
<E> SimpleSaveResult<E> save(Input<E> input);
```

As you see, different input type cause the sample result type. So these method cannot be merged into one, and the first method checks the input argument must not be `Input` at runtime.

However, for `saveAll`
```
<E> BatchSaveResult<E> saveAll(Collection<E> entities);
<E> BatchSaveResult<E> saveAll(Collection<Input<E>> inputs);
```
will cause compilation error. Be different with languages support real generic type such as C++/C#, Java erase the generic type so both of these two methods have the same signature whose parametere is `Collection`.

Solution1, don't affect the existing users
```
<E> BatchSaveResult<E> saveAll(Collection<E> entities);
<E> BatchSaveResult<E> saveAllInputs(Collection<Input<E>> inputs);
```

Solution2, change the API but affect exsiting users.

```
<E> BatchSaveResult<E> saveEntities(Collection<E> entities);
<E> BatchSaveResult<E> saveInputs(Collection<Input<E>> inputs);
```

**Comment by ArgonarioD** [comment]
saveAll保留，但标记 `@Deprecated` ，以后版本删除。
添加2的实现。
顺便建议记录每个版本破坏性更新的迁移手册（）

**Comment by pot-mot** [comment]
Solution2, 变更 API 以保证明确
@Deprecated saveAll，0.9 移除

**Comment by qinkangdeid** [comment]
Solution2

**Comment by leeaee** [comment]
Solution2
